### PR TITLE
[RHPAM-3706] Use BOM managed version for non-productized quarkus-juni… [1.5.x]

### PR DESCRIPTION
**Pull request by AntStephenson**
…t5-internal and add new prop for non-bom managed dep.

When aligning internally to a productized version of Quarkus in [RHPAM-3706](https://issues.redhat.com/browse/RHPAM-3706) we realised there are some deps that aren't part of the runtime and therefore aren't productized, we use `quarkus-junit5-internal` which is defined in the quarkus-bom so this change uses the BOM managed version (as is done in other places).

There is also quarkus-test-maven being used that is not BOM managed, as a temporary solution to keep it at the community version I'm introducing a version property to be kept in sync with version.io.quarkus that would not be touched by PME when aligning the version in the productized build.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-build/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by AntStephenson** [comment]
Jenkins retest this

**Comment by AntStephenson** [comment]
Jenkins retest this

**Comment by AntStephenson** [comment]
@danielezonca @radtriste Hi could you review this please, required for 7.11.1 to avoid manual patching on the prod side.

@danielezonca suggested aligning to ${version.io.quarkus} for the new version prop but PME won't touch it otherwise (see [here](https://github.com/kiegroup/kogito-runtimes/pull/1378#discussion_r660611626))

File: kogito-build-parent/pom.xml [hunk 1]
```diff
@@ -117,6 +117,8 @@
 
     <!-- this version property is used in plugins but also in dependencies too -->
     <version.io.quarkus>1.11.6.Final</version.io.quarkus>
+    <!-- non-productized non-bom managed dependency keep in sync with version.io.quarkus -->
+    <version.io.quarkus.test-maven>1.11.6.Final</version.io.quarkus.test-maven>
```
Comment (Ginxo) [inline kogito-build-parent/pom.xml:121]:
```suggestion
    <version.io.quarkus.test-maven>${version.io.quarkus}</version.io.quarkus.test-maven>
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
can you please check my suggestion? thanks @AntStephenson

File: kogito-build-parent/pom.xml (hunk 1 above)
Comment (AntStephenson) [reply kogito-build-parent/pom.xml:121]: See discussion on master PR :)

**Pull request merged**